src/promptflow/tests/sdk_cli_test/unittests/test_utils.py

```diff
@@ -202,18 +205,30 @@ def test_concurrent_execution_of_refresh_connections_dir(self, concurrent_count)
         for thread in threads:
             thread.join()
 
-    @pytest.mark.parametrize("concurrent_count", [1, 2, 4, 8])
-    def test_concurrent_hint_for_update(self, concurrent_count):
-        from concurrent.futures import ThreadPoolExecutor
-        with patch('promptflow._utils.version_hint_utils.datetime') as mock_datetime:
-            mock_datetime.datetime.now.return_value = datetime.datetime.now()
-            mock_datetime.datetime.strptime.return_value = datetime.datetime.now() - datetime.timedelta(days=8)
+    def test_concurrent_hint_for_update(self):
+        def mock_check_latest_version():
+            time.sleep(5)
+            check_latest_version()
+
+        with patch('promptflow._utils.version_hint_utils.datetime') as mock_datetime, patch(
+                "promptflow._utils.version_hint_utils.check_latest_version", side_effect=mock_check_latest_version):
+            from promptflow._sdk._telemetry import monitor_operation
+
+            class HintForUpdate:
+                @monitor_operation(activity_name="pf.flows.test")
```

Comment:
Is this method simpler? Is it necessary for us to use class?
`wrapper_fun = monitor_operation(...)(function)`

Reply:
it will raise error: `TypeError: wrapper() missing 1 required positional argument: 'self'` if not use class